spotlight-client/src/charts/WindowedTimeSeries.tsx

```diff
@@ -137,6 +145,93 @@ const WindowedTimeSeries: React.FC<{
+                tickFormat: formatAsNumber,
+                tickSize: 0,
+              },
+              {
+                orient: "bottom",
+                tickFormat: (time: Date) =>
+                  time.getDate() === 1 ? format(time, "MMM y") : null,
+                ticks: isMobile ? 5 : 10,
+                tickSize: 0,
+              },
+            ]}
+            baseMarkProps={BASE_MARK_PROPS}
+            lines={data}
+            lineStyle={(d: DataSeries) => ({
+              fill:
+                highlighted && highlighted.label !== d.label
+                  ? highlightFade(d.color)
+                  : d.color,
+              stroke: colors.background,
+              strokeWidth: 1,
+            })}
+            // @ts-expect-error Semiotic typedefs are incomplete, this works
+            lineType={{ type: "stackedarea", sort: null }}
+            // @ts-expect-error Semiotic typedefs are wrong, can be true for default matte
+            matte
+            minimap={{
+              // @ts-expect-error Semiotic typedefs are incomplete
+              axes: [],
+              baseMarkProps: BASE_MARK_PROPS,
+              brushEnd: (brushExtent: Date[]) => {
+                const [start, end] = brushExtent || [];
+                if (start && end) {
+                  if (isNewRange({ start, end })) {
+                    setTimeRangeId("custom");
+                  }
+
+                  setDateRangeStart(new Date(start));
+                  setDateRangeEnd(new Date(end));
+                }
+              },
+              margin: { ...MARGIN, bottom: 0 },
+              size: [width, MINIMAP_HEIGHT],
+              // suppress brush area if width is 0 to prevent SVG rendering errors
+              xBrushable: Boolean(width),
+              xBrushExtent: [dateRangeStart, dateRangeEnd],
+              yBrushable: false,
+            }}
+            pointStyle={{ display: "none" }}
+            xAccessor="date"
+            yAccessor="count"
+          />
+        );
+
+        const vizNoMinimap = (
+          <XYFrame
+            axes={[
+              {
+                orient: "left",
+                tickFormat: formatAsNumber,
+                tickSize: 0,
+              },
+            ]}
+            baseMarkProps={BASE_MARK_PROPS}
+            lines={data}
+            lineStyle={(d: DataSeries) => ({
+              fill:
+                highlighted && highlighted.label !== d.label
+                  ? highlightFade(d.color)
+                  : d.color,
+              stroke: colors.background,
+              strokeWidth: 1,
+            })}
+            // @ts-expect-error Semiotic typedefs are incomplete, this works
+            lineType={{ type: "stackedarea", sort: null }}
+            // @ts-expect-error Semiotic typedefs are wrong, can be true for default matte
+            matte
+            pointStyle={{ display: "none" }}
+            xAccessor="date"
+            yAccessor="count"
```

Comment:
I think by doing it this way we are creating a bunch of expensive React and Semiotic stuff that we don't intend to render, as well as defining most of the props redundantly? 

Rather than duplicating all these props, we should just collect them in a base object that we extend as needed with the additional props needed for the minimap. And instead of instantiating both Frame components we should just conditionally pick one ... could either store that as a variable or conditionally render `{preview ? <XYFrame {...chartProps} /> : <MinimapXYFrame {...chartProps} />}` instead of what's currently there.